### PR TITLE
[SIP-15] Making client time use UTC as the local time

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Whilst playing with the time range picker I discovered that for relative times ("Last Week" etc.) the reference time (either shown in default or via a tooltip) would use the client timezone rather than the server timezone. This was problematic say at 2019-10-24:18:00:00 PDT the UI would reference `2019-10-24 00:00:00` in various tooltips however when the query was executed it would use `2019-10-25 00:00:00`.  

Initially I was unsure whether we should parse the server timezone to the client but it seems in [other places](https://github.com/apache/incubator-superset/blob/bd9a2c15e79f2f337bf3bdfd67603eee8d0e16a9/superset/assets/src/modules/dates.js#L27) there's the assumption that times are expressed in UTC (irrespective of the server timezone). 

Note the time range picker does not explicitly set the timezone by default for absolute dates, i.e., the query will actually run in reference to the timezone of the executing database engine. However one can specify the timezone via the ISO 8601 format. I’m not certain whether Superset should be more opinionated with regards to timezones. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

I added a tooltip to explicitly define where/how times are evaluated. 

Screen Shot 2019-10-26 at 6 17 08 PM

### TEST PLAN

CI. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue: https://github.com/apache/incubator-superset/issues/6360
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @betodealmeida @etr2460 @mistercrunch 
cc: @vylc

### CODE REVIEW

**Comment by villebro** [comment]
While I agree UTC is the natural starting point for `now`, there will be many people relying on the current behaviour. So there should at least be a mention in `UPDATING.md` informing users of the change. Furthermore, and added bonus would be making this optional, i.e. being able to choose whether or not the reference point is local or UTC time, but that would require more backend -> frontend plumbing, so might cause unnecessary clutter in the codebase.

**Comment by john-bodley** [comment]
@villebro I agree regarding adding a section to `UPDATING.md`. The timezone logic is possibly more nuances especially if we’re talking about non-explicit timezone formats, i.e., not epoch which is in reference to UTC.

There are three different clocks:

1. The database engine.
2. The Superset server.
3. The Superset client.

and thus unless ISO 8601 times are encoded with the timezone (which can be specified in the time range picker) the time logic can become quite complex especially if all three clocks are not using the same timezone.

Currently for the time range picker absolute times are specified using the client clock (or per this PR use UTC as the local time) without a designated timezone. Similarly relative times are evaluated on the Superset server using the server clock without a designated timezone. When the query is run the times are in reference to the local time of the database engine. This can all get quite messy if all three clocks are not using the same timezone.

File: superset/assets/src/explore/controls.jsx [hunk 1]
```diff
@@ -951,6 +951,7 @@ export const controls = {
     freeForm: true,
     label: t('Time range'),
     default: t('Last week'),
+    description: t('The time range for the visualization. Client times are expressed using UTC as the local time.'),
```
Comment (etr2460) [inline superset/assets/src/explore/controls.jsx]: wording nit: I prefer `The time range for the visualization. Times are in UTC`. Gets the same point across but is much more succinct

**Comment by john-bodley** [comment]
@villebro I addressed your comments and updated the tooltip to explicitly mentioned where times were evaluated and how timezones were used.

File: UPDATING.md [hunk 2]
```diff
@@ -23,6 +23,9 @@ assists people when migrating to a new version.
 
 ## Next Version
 
+* [8450](https://github.com/apache/incubator-superset/pull/8450): The time ranger picker
+now uses UTC for the tooltips and default placeholder timestamps (san timezone).
```
Comment (etr2460) [inline UPDATING.md]: spelling nit: sans

**Review by etr2460 — APPROVED** [review]
lgtm with one nit

**Pull request merged**